client/src/sections/courses/LearningSection/AddLearnerModal.tsx

```diff
@@ -1,125 +1,177 @@
+/* eslint-disable multiline-ternary */
+import React, { useState } from 'react';
+import { useGetEnrolleesQuery } from '@/services/enrolleeAPI';
 import Button from '@/src/shared/components/Button';
 import Modal from '@/src/shared/components/Modal/Modal';
 import Pagination from '@/src/shared/components/Pagination';
 import SearchBar from '@/src/shared/components/SearchBar/SearchBar';
 import CloseIcon from '@/src/shared/icons/CloseIcon';
-import { type Learner } from '@/src/shared/utils';
-import React, { useState } from 'react';
-
-interface AddLearnerModalProps {
-  learners: Learner[];
-}
+import { useRouter } from 'next/router';
+import { useDispatch } from 'react-redux';
+import type { AppDispatch } from '@/app/store';
+import { storeEnrollee } from '@/services/posts/addNewEnrollees';
 
-const AddLearnerModal: React.FC<AddLearnerModalProps> = ({ learners }: AddLearnerModalProps): JSX.Element => {
+const AddLearnerModal: React.FC = (): JSX.Element => {
+  const router = useRouter();
+  const params = router.query;
   const [currentPage, setCurrentPage] = useState(1);
   const [isModalOpen, setIsModalOpen] = useState(false);
+  const [search, setSearch] = useState('');
+  const [page, setPage] = useState(1);
+  const [selectedIds, setSelectedIds] = useState<number[]>([]);
+  const courseID = params.id;
+  const dispatch = useDispatch<AppDispatch>();
+  const { data, refetch } = useGetEnrolleesQuery({
+    courseID,
+    search,
+    page,
+  });
 
-  const handleAddLearners = (): void => {
-    alert('Add Learners');
-  };
+  const enrollees = data?.enrollees;
 
   const handleModalClose = (): void => {
+    setSelectedIds([]);
     setIsModalOpen(false);
+    setSearch('');
+    setPage(1);
+    void refetch();
```

Comment:
```suggestion
    setPage(1);
```

1. You do not need to use `refetch()` in your case, because you are already changing search property you given in `useGetEnrolleesQuery `, basically, when that properties are changed, rtk query checks if its different than before, and also looks for cached data that matches the props, if its not found,  then it makes a request. That way there wont be any duplicated requests

but in your code, you are forcing rtk to refetch the data, and it makes duplicate request even if there is already the data exist in cache

Please refer to docs for caching behaviour: https://redux-toolkit.js.org/rtk-query/usage/cache-behavior



2. Please make sure setPage(1) is actually functioning as intended, I've tried removing it on my end, and it made no difference

Please apply these to all related